### PR TITLE
TF pin for 11/14/2022

### PR DESCRIPTION
**Pull request by ManfeiBai**
Tensorflow pin update to
```
commit 8b2e0ca8e3fed42abfaa7baeabe25fc8dfd1c2ab (HEAD)
Author: Rahul Batra <rahbatra@amd.com>
Date:   Mon Nov 14 15:22:48 2022 +0000

    [ROCm]: Updated CI files to ROCm5.3
```

### CODE REVIEW

File: torch_xla/csrc/layout_manager.cpp [hunk 1]
```diff
@@ -120,7 +120,7 @@ xla::Shape MakeShapeWithSortedLayout(absl::Span<const int64_t> dimensions,
   std::sort(layout.begin(), layout.end(), [&](int64_t a, int64_t b) {
     return dimensions[a] > dimensions[b];
   });
-  return xla::ShapeUtil::MakeShapeWithLayout(type, dimensions, layout);
+  return xla::ShapeUtil::MakeShapeWithDenseLayout(type, dimensions, layout);
```
Comment (yeounoh) [inline torch_xla/csrc/layout_manager.cpp:123]: This is correct in our case.

File: third_party/xla_client/pjrt_computation_client.cc [hunk 2]
```diff
@@ -60,7 +60,7 @@ PjRtComputationClient::PjRtComputationClient() {
     client_ = xla::GetTpuClient(max_inflight_computations).value();
   } else if (device_type == "TPU_C_API") {
     TF_VLOG(1) << "Initializing PjRt C API client...";
-    client_ = std::move(xla::GetCApiClient().value());
+    client_ = std::move(xla::GetCApiClient(device_type).value());
```
Comment (yeounoh) [inline third_party/xla_client/pjrt_computation_client.cc:63]: I think the canonical device type should be "TPU" instead of "TPU_C_API"

**Review by yeounoh — CHANGES_REQUESTED** [review]
Mostly LGTM -- thanks and great work @ManfeiBai . Left some comments for you to take a look. We also need to update setup.py for libtpu ver. Once we can do a quick training test on a TPUVM we can merge this :)

File: third_party/xla_client/pjrt_computation_client.cc (hunk 2 above)
Comment (JackCaoG) [reply third_party/xla_client/pjrt_computation_client.cc:63]: `TPU_C_API` is a special mode to test the new CAPI bridge.

File: tf_patches/f16_abi_clang.diff [hunk 3]
```diff
@@ -1,12 +1,11 @@
-upstream CI will fail without this
```
Comment (JackCaoG) [inline tf_patches/f16_abi_clang.diff:1]: can we restore this line.

File: third_party/xla_client/pjrt_computation_client.cc (hunk 2 above)
Comment (yeounoh) [reply third_party/xla_client/pjrt_computation_client.cc:63]: `device_type` check should be against "TPU_C_API", but the call to `xla::GetCApiClient()` I think we should pass "TPU".

**Comment by JackCaoG** [comment]
we need to find a way to drop all these diffs, maybe a good topic for next week's fixit.

**Comment by yeounoh** [comment]
> we need to find a way to drop all these diffs, maybe a good topic for next week's fixit.

Yea, it would be nice if our build works straight with the TF without us making any local changes. Having said that, I reviewed the diffs with Manfei, they are all needed this time. But, I share the same concern with you and +1.

File: torch_xla/csrc/layout_manager.cpp (hunk 1 above)
Comment (ManfeiBai) [reply torch_xla/csrc/layout_manager.cpp:123]: Thanks, Yeounoh!

**Comment by JackCaoG** [comment]
> > we need to find a way to drop all these diffs, maybe a good topic for next week's fixit.
> 
> Yea, it would be nice if our build works straight with the TF without us making any local changes. Having said that, I reviewed the diffs with Manfei, they are all needed this time. But, I share the same concern with you and +1.

I think we need to resolve this from the google3 tf side, basically to upstream some of the diffs.

File: third_party/xla_client/pjrt_computation_client.cc (hunk 2 above)
Comment (ManfeiBai) [reply third_party/xla_client/pjrt_computation_client.cc]: Thanks, updated to "TPU"

File: tf_patches/f16_abi_clang.diff (hunk 3 above)
Comment (ManfeiBai) [reply tf_patches/f16_abi_clang.diff]: Thanks, updated ~

File: third_party/xla_client/pjrt_computation_client.cc (hunk 2 above)
Comment (yeounoh) [reply third_party/xla_client/pjrt_computation_client.cc]: Yea, for a future reference, tpu initializer would do something like,
```
stream_executor::tpu::SetPjrtApi("TPU", GetTpuPjrtApi()).IgnoreError();
```
and the "TPU" is used as a key (lower-cased) to map the C API to the device type. cc @will-cromar @JackCaoG

Comment (ManfeiBai) [reply third_party/xla_client/pjrt_computation_client.cc]: should we lower-case "TPU" right now or keep "TPU" and lower-cased by function?

Comment (yeounoh) [reply third_party/xla_client/pjrt_computation_client.cc]: It's done by the function.

Comment (ManfeiBai) [reply third_party/xla_client/pjrt_computation_client.cc]: Thanks, Yeounoh!

**Comment by ManfeiBai** [comment]
I ran `git reset --hard 59e77b15649dbf54d6eb26de44f4da0185431033` in TPUVM to pull the newest modify in this pr, but the newest tf commit have not been pulled automatically in TPUVM; However the tensorflow newest commit has been pushed, and is visiable under "File Changed" tab now;
How do we make sure the final "third_party/tensorflow" is the newest commit in github?

**Comment by JackCaoG** [comment]
your brnach's tf is pin to ~14 days ago https://github.com/tensorflow/tensorflow/tree/8b2e0ca8e3fed42abfaa7baeabe25fc8dfd1c2ab.

If that's where you want to pin, you just need to do a `git submodule sync` and `git submodule update --init --recursive` on your TPUVM. @yeounoh Are we doing another one for Dec? If not maybe it is a good idea to sync to Dec 1's head since next update will be Jan.

**Review by miladm — APPROVED** [review]
LGTM after the conflict resolution and retest of the CI.

Did you verify the upstream pytorch tests pass with this branch?

**Comment by yeounoh** [comment]
> your brnach's tf is pin to ~14 days ago https://github.com/tensorflow/tensorflow/tree/8b2e0ca8e3fed42abfaa7baeabe25fc8dfd1c2ab. (by clicking https://github.com/ManfeiBai/PyTorchXLA/tree/fortfpin11222022/third_party)
> 
> If that's where you want to pin, you just need to do a `git submodule sync` and `git submodule update --init --recursive` on your TPUVM. @yeounoh Are we doing another one for Dec? If not maybe it is a good idea to sync to Dec 1's head since next update will be Jan.

Yea, @ManfeiBai is repurposing this with a Dec commit (from 12/2/22). @miladm yea, we will test this on TPUVM & upstream. This is still WIP.

**Comment by miladm** [comment]
FWIW, there was a recent dynamic shape refactor in tensorflow repo that may influence our DS initiative. Wondering if we should squeeze that in - if not too late. If so, happy to share more context. Wdyt?

**Comment by yeounoh** [comment]
> FWIW, there was a recent dynamic shape refactor in tensorflow repo that may influence our DS initiative. Wondering if we should squeeze that in - if not too late. If so, happy to share more context. Wdyt?

Yea, let me know of the date, @miladm I will work with @ManfeiBai to squeeze that in.

**Comment by ManfeiBai** [comment]
Thanks for the comments, @yeounoh , @miladm , @JackCaoG ,
The new TF pin update with the Dec commit (from 12/2/22) is in https://github.com/pytorch/xla/pull/4324,
We close this PR because we will skip 11/14/22 and update TF pin to 12/2/22 directly, and welcome to continue our discuss in #4324 too!

**Pull request closed without merging**